### PR TITLE
LPS-204967 Avoid new-added unused @reference variable

### PR DESCRIPTION
**Pull request by qz1010**

### CODE REVIEW

**Comment by qz1010** [comment]
ci:test:sf

**Comment by qz1010** [comment]
ci:test:relevant

**Comment by JanisZhang** [comment]
@qz1010 Thanks for working on this!

**Comment by ling-alan-huang** [comment]
see https://github.com/ling-alan-huang/liferay-portal/pull/2388

**Comment by ling-alan-huang** [comment]
@qz1010 I am not seeing how do we avoid new-added unused variable in this check.

File: source-formatter.properties [hunk 1]
```diff
@@ -390,6 +390,7 @@
         modules/apps/calendar/calendar-web,\
         modules/apps/change-tracking/change-tracking-rest-impl,\
         modules/apps/change-tracking/change-tracking-service,\
+        modules/apps/change-tracking/change-tracking-service,\
```
Comment (ling-alan-huang) [inline source-formatter.properties]: Duplicates?

File: source-formatter.properties [hunk 2]
```diff
@@ -423,6 +424,7 @@
         modules/apps/dispatch/dispatch-service,\
         modules/apps/document-library/document-library-repository-portlet-file-repository-impl,\
         modules/apps/document-library/document-library-service,\
+        modules/apps/document-library/document-library-service,\
```
Comment (ling-alan-huang) [inline source-formatter.properties]: ???

File: source-formatter.properties [hunk 3]
```diff
@@ -461,26 +466,37 @@
         modules/apps/layout/layout-reports-web,\
         modules/apps/layout/layout-seo-service,\
         modules/apps/layout/layout-taglib,\
+        modules/apps/layout/layout-type-controller/layout-type-controller-display-page,\
         modules/apps/message-boards/message-boards-comment,\
         modules/apps/message-boards/message-boards-moderation,\
         modules/apps/message-boards/message-boards-service,\
         modules/apps/oauth2-provider/oauth2-provider-service,\
+        modules/apps/oauth2-provider/oauth2-provider-service,\
```
Comment (ling-alan-huang) [inline source-formatter.properties]: ??

**Comment by ling-alan-huang** [comment]
Check again before sending PR.

**Comment by qz1010** [comment]
> @qz1010 I am not seeing how do we avoid new-added unused variable in this check.
Only in allowedUnusedReferenceVariableDirNames paths allows unused variables containing @Refernece to appear. Such variables cannot appear in new paths in the future.

**Comment by qz1010** [comment]
see #2390

**Pull request closed without merging**